service/userService.go

```diff
@@ -1,20 +1,70 @@
 package service
 
-import "github.com/eduboard/backend"
+import (
+	"errors"
+	"fmt"
+	"github.com/eduboard/backend"
+)
 
 type UserService struct {
 	r eduboard.UserRepository
+	a eduboard.UserAuthenticator
+	tG eduboard.UserTokenGenerator
 }
 
-func NewUserService(repository eduboard.UserRepository) *UserService {
+func NewUserService(repository eduboard.UserRepository, authenticator eduboard.UserAuthenticator, tokenGenerator eduboard.UserTokenGenerator) *UserService {
```

Comment:
I think we don't need to pass these as initialisers for the service. They should definitely be here, but they could just as well be defined below.
I think the reason you did this was because I said that "communication will be done exclusively through interfaces". What I meant by that was communication across the different layers: Domain (User, Course), Service (Services) and Infrastructure (http and mongodb). Sorry for not being clear enough on that.
In cases like this I think we can safely assume that authentication is an important part of the userService and can be integrated as such.